pandas/tests/indexes/categorical/test_category.py

```diff
@@ -975,3 +975,9 @@ def test_engine_type(self, dtype, engine_type):
             ci.values._codes = ci.values._codes.astype("int64")
         assert np.issubdtype(ci.codes.dtype, dtype)
         assert isinstance(ci._engine, engine_type)
+
+    def test_getitem_2d_deprecated(self):
+        # GH#30588 multi-dim indexing is deprecated, but raising is also acceptable
+        idx = self.create_index()
+        with pytest.raises(ValueError, match="cannot mask with array containing NA"):
```

Comment:
We shouldn't test wrong error messages ...

Reply:
This may be a bug in how we're handling tuples. We convert the tuple `(slice(None), None)` to an ndarray

```python
array([slice(None, None, None), None], dtype=object)
```

which gets interpreteded as a boolean mask with missing values, and so raise.

I'm going to update the test to have a 2d ndarray as the indexer, nad open an issue for the bug.